apps/tmbitwarden.sh

```diff
@@ -61,39 +61,154 @@ msg_box "OK, please do the necessary preparations before you run this script and
+    LogLevel warn
+    CustomLog ${APACHE_LOG_DIR}/access.log combined
+    ErrorLog ${APACHE_LOG_DIR}/error.log
+    # Just in case - see below
+    SSLProxyEngine On
+    SSLProxyVerify None
+    SSLProxyCheckPeerCN Off
+    SSLProxyCheckPeerName Off
+    # contra mixed content warnings
+    RequestHeader set X-Forwarded-Proto "https"
+    # basic proxy settings
+    ProxyRequests off
+    ProxyPassMatch (.*)(\/websocket)$ "ws://127.0.0.1:5178/$1$2"
+    ProxyPass / "http://127.0.0.1:5178/"
+    ProxyPassReverse / "http://127.0.0.1:5178/"
+        
+    <Location />
+        ProxyPassReverse /
+    </Location>
+</VirtualHost>
+HTTPS_CREATE
+
+    if [ -f "$HTTPS_CONF" ];
+    then
+        print_text_in_color "$IGreen" "$HTTPS_CONF was successfully created."
+        sleep 1
+    else
+        print_text_in_color "$IRed" "Unable to create vhost, exiting..."
+        print_text_in_color "$IRed" "Please report this issue here $ISSUES"
+        exit 1
+    fi
+fi
+
+# Install certbot (Let's Encrypt)
+install_certbot
+
+# Generate certs and  auto-configure  if successful
+if generate_cert  "$SUBDOMAIN"
+then
+    # Generate DHparams chifer
+    if [ ! -f "$DHPARAMS_SUB" ]
+    then
+        openssl dhparam -dsaparam -out "$DHPARAMS_SUB" 4096
+    fi
+    print_text_in_color "$IGreen" "Certs are generated!"
+    a2ensite "$SUBDOMAIN.conf"
+    restart_webserver
 else
-    sleep 0.1
+    last_fail_tls "$SCRIPTS"/apps/tmbitwarden.sh
 fi
 
 # Install Docker
 install_docker
 install_if_not docker-compose
 
-# Stop Apache to not conflict when LE is run
-check_command systemctl stop apache2.service
-
 # Install Bitwarden 
 install_if_not curl
 cd /root
 curl_to_dir "https://raw.githubusercontent.com/bitwarden/core/master/scripts" "bitwarden.sh" "/root"
+export DOMAIN="$SUBDOMAIN"
 chmod +x /root/bitwarden.sh
 check_command ./bitwarden.sh install
-sed -i "s|http_port.*|http_port: 8080|g" /root/bwdata/config.yml
-sed -i "s|https_port.*|https_port: 8443|g" /root/bwdata/config.yml
+sed -i "s|http_port.*|http_port: 5178|g" /root/bwdata/config.yml
+sed -i "s|https_port.*|https_port: 5179|g" /root/bwdata/config.yml
+sed -i "s|^url: .*|url: https://$SUBDOMAIN|g" /root/bwdata/config.yml
+sed -i 's|http://|https://|g' /root/bwdata/env/global.override.env
+
+# add certificate files to config.yml
+sed -i "s|^ssl_certificate_path:.*|ssl_certificate_path: $CERTFILES/$SUBDOMAIN/fullchain.pem|g" /root/bwdata/config.yml
+sed -i "s|^ssl_key_path:.*|ssl_key_path: $CERTFILES/$SUBDOMAIN/privkey.pem|g" /root/bwdata/config.yml
+sed -i "s|^ssl_diffie_hellman_path:.*|ssl_diffie_hellman_path: $CERTFILES/$SUBDOMAIN/dhparam.pem|g" /root/bwdata/config.yml
```

Comment:
This should theoratically work now, but I think we don't need this at all since we are using the apache2 reverse proxy. 
It probably never gets into action because of our reverse proxy.
So I don't get why you are trying so hard to get this right?

Reply:
It hit me later than it would never work as the redirection wouldn't work anyway.

Let's continue in the other PR as you proposed.